### PR TITLE
Allow Vector{Symbol} in some ring constructors

### PR DESCRIPTION
**Pull request by fingolfin**
This avoids some needless conversions from Symbol to String and back.

### CODE REVIEW

**Comment by tthsqe12** [comment]
I will just complete this for Weyl algebras, add some tests, and then merge it.

**Comment by fingolfin** [comment]
Thank you

File: docs/src/noncommutative.md [hunk 1]
```diff
@@ -96,11 +96,11 @@ julia> Q, (x, y) = QuotientRing(G, std(I))
 ### WeylAlgebra
 
 ```julia
-function WeylAlgebra(R::Union{Ring, Field}, s::Vector{String};
+function WeylAlgebra(R::Union{Ring, Field}, s::Union{Vector{String}, Vector{Symbol}};
                      ordering = :degrevlex, ordering2::Symbol = :comp1min,
                      cached::Bool = true, degree_bound::Int = 0)
 
-function WeylAlgebra(R::Union{Ring, Field}, s::Matrix{String};
+function WeylAlgebra(R::Union{Ring, Field}, s::Union{Matrix{String}, Vector{Symbol}};
```
Comment (fingolfin) [inline docs/src/noncommutative.md:103]: Do you mean this?
```suggestion
function WeylAlgebra(R::Union{Ring, Field}, s::Union{Matrix{String}, Matrix{Symbol}};
```

Comment (tthsqe12) [reply docs/src/noncommutative.md:103]: Huh, yes, but weird that the test code still passes. Something funny is going on.

Comment (tthsqe12) [reply docs/src/noncommutative.md:103]: Oops. Didn't have Singular.jl dev`d.

**Pull request merged**